src/zenml/orchestrators/step_run_utils.py

```diff
@@ -382,3 +388,27 @@ def link_output_artifacts_to_model_version(
                 artifact_version=output_artifact,
                 model_version=model_version,
             )
+
+
+def cascade_tags_for_output_artifacts(
+    artifacts: Dict[str, List[ArtifactVersionResponse]],
+    tags: Optional[List[Union[str, Tag]]] = None,
+) -> None:
+    """Tag the outputs of a step run.
+
+    Args:
+        artifacts: The step output artifacts.
+        tags: The tags to add to the artifacts.
+    """
+    if tags is None:
+        return
+
+    for output_artifacts in artifacts.values():
+        for output_artifact in output_artifacts:
+            cascade_tags = [
```

Comment:
This could be done outside the double for loop so it happens just once maybe?

Reply:
Done.